sklearn/tree/_tree.pyx

```diff
@@ -91,22 +91,29 @@ cdef class TreeBuilder:
         """Build a decision tree from the training set (X, y)."""
         pass
 
-    cdef inline _check_input(self, object X, cnp.ndarray y,
-                             cnp.ndarray sample_weight):
+    cdef inline _check_input(
+        self,
+        object X,
+        cnp.ndarray y,
+        cnp.ndarray sample_weight
+    ):
         """Check input dtype, layout and format"""
         if issparse(X):
+            # csc is fortran by default
             X = X.tocsc()
             X.sort_indices()
 
+            # if we need to copy to convert the datatype, then do it using fortran
             if X.data.dtype != DTYPE:
-                X.data = np.ascontiguousarray(X.data, dtype=DTYPE)
+                X.data = np.asfortranarray(X.data, dtype=DTYPE)
```

Comment:
I do think think this makes a difference, because `X.data` is 1D. Since the code does the same thing, I prefer to revert to keep the `git blame` .

Reply:
Done